File: src/main/java/com/dv/telegram/config/BotSetting.java
package com.dv.telegram.config;

import java.util.List;

public interface BotSetting<T> {
    String getName(); // english name, no spaces
    String getDescription(); // description, in russian

    T getValue();
    void setValue(String value);

    static List<BotSetting<?>> getAllSettings() {
        return List.of(
            new StartMessage(),
            new DeleteBotCallMessageOnMessageReply(),
            new BotTriggerMode(),
            new ReplyWhenNoAnswer(),
            new NoAnswerReply()
        );
    }
}

File: src/main/java/com/dv/telegram/config/BotSetting.kt
package com.dv.telegram.config

sealed interface BotSetting<T> {
    val name: String // English name, no spaces

    val description: String // description, in Russian
    fun getValue(): T
    fun setValue(value: String)

    companion object {
        val allSettings = listOf(
            StartMessage(),
            DeleteBotCallMessageOnMessageReply(),
            BotTriggerMode(),
            ReplyWhenNoAnswer(),
            NoAnswerReply()
        )
    }
}
